apps/docs-snippets2/scripts/wrap-snippets.ts

```diff
@@ -0,0 +1,38 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { globSync } from 'glob';
+
+const importsReg = /import[\s\S]+from.+['"];/gm;
+
+const wrapper = `export const main = async (): Promise<any | any[]> => {
+  const logs: any[] = [];
+  const console = { log (...args: any[]) { logs.push(args) }};
+  // ———>>>
+  %SNIPPET%
+  // <<<———
+  const singleCall = logs.length === 1 && logs[0].length === 1;
+  return singleCall ? logs[0][0] : logs;
+}`.replace(/^\s{4}/gm, '  ');
```

Comment:
All code snippets are wrapped inside a `main` function and saved to a `.wrapped.ts` equivalent.

The params for all `console.log()` calls inside the snippet are collected and can be later asserted.